sheetstorm/management/commands/generate_test_data.py

```diff
@@ -40,6 +37,51 @@ class Command(BaseCommand):
 
     @transaction.atomic
     def handle(self, *args: Any, **options: Union[bool, int, None]) -> None:
+        is_request_to_use_prepared_set = self._get_request_to_create_data_using_prepared_set(options)
+
+        if is_request_to_use_prepared_set:
+            self.create_data_using_prepared_set(options)
+        else:
+            self.execute_creating_users(options)
+            self.execute_creating_project(options)
+
+        logging.info(f"Total number of users in the database: {CustomUser.objects.count()}")
+        logging.info(f"Total number of projects in the database: {Project.objects.count()}")
+
+    @staticmethod
+    def _get_request_to_create_data_using_prepared_set(options: OptionsDictType) -> Any:
+        return any(
+            [
+                options[TestDataSize.SMALL.name],
+                options[TestDataSize.MEDIUM.name],
+                options[TestDataSize.LARGE.name],
+                options[TestDataSize.EXTRA_LARGE.name],
+            ]
+        )
+
+    def create_data_using_prepared_set(self, options: OptionsDictType) -> None:
+        requested_set = self._pick_dataset_to_create(options)
+
+        self.execute_creating_users(requested_set)
+        self.execute_creating_project(requested_set)
+
+    @staticmethod
+    def _pick_dataset_to_create(options: OptionsDictType) -> Any:
+        set_sizes_list = [
+            TestDataSize.SMALL.name,
+            TestDataSize.MEDIUM.name,
+            TestDataSize.LARGE.name,
+            TestDataSize.EXTRA_LARGE.name,
+        ]
+        for set_size in set_sizes_list:
+            if options[set_size]:
+                requested_set = DATA_SETS[set_size]
+                break
+        else:
+            raise NotAnySetRequestedException("Not any set requested")
+        return requested_set
```

Comment:
This method is too complicated, you need to simplify it. This is wrong that you're iterating here element by element for list of data sizes. Please start from adding init method for all variables and then you won't need `options` here because you all will have accessible from instance